pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -171,29 +179,72 @@ func newReconciler(ofClient openflow.Client, ifaceStore interfacestore.Interface
 // invoke the add or update method accordingly.
 func (r *reconciler) Reconcile(rule *CompletedRule) error {
 	klog.Infof("Reconciling rule %s of NetworkPolicy %s/%s", rule.ID, rule.PolicyNamespace, rule.PolicyName)
+	var err error
+	var ofPriority *uint16
 
 	value, exists := r.lastRealizeds.Load(rule.ID)
+	r.priorityMutex.Lock()
+	defer func() {
+		// For CNP, only release priorityMutex after rule is installed on OVS. Otherwise,
+		// priority re-assignments for flows that has already been assigned priority but
+		// not yet installed on OVS will be missed.
+		if err != nil || ofPriority != nil {
+			r.priorityMutex.Unlock()
+		}
+	}()
+	ofPriority, err = r.getOFPriority(rule)
+	if err != nil {
+		return err
+	}
+	// Release the priorityMutex immdiately for k8s NP
```

Comment:
Could we check K8s NP earlier with rule itself and not to lock at all?

Reply:
Done.